### ISSUE TITLE
`clear_cache` doesn't work after one of the tables has been opened in the same R session

### ISSUE THREAD

**Issue by sbashevkin** [Delta-Stewardship-Council/deltafish#9]
After opening a table with `open_fish` or `open_survey` you can't clear the cache with `clear_cache` unless you restart R.

It might be helpful to add a trycatch or similar error catching system to return a helpful error to users telling them to clear their environment, restart R, and then run `clear_cache`

Here is the warning message that is returned:
```deltafish:::clear_cache_f(cache_dir)
Removing existing cache.
NULL
Warning messages:
1: In FUN(X[[i]], ...) :
  cannot remove file 'C:\Users\SBASHE~1\AppData\Local/LTMRdata-test/LTMRdata-test/Cache/fish/Taxa=Acipenser medirostris/part-0.parquet', reason 'Permission denied'
2: In FUN(X[[i]], ...) :
  cannot remove file 'C:\Users\SBASHE~1\AppData\Local/LTMRdata-test/LTMRdata-test/Cache/fish/Taxa=Acipenser transmontanus/part-0.parquet', reason 'Permission denied'
3: In FUN(X[[i]], ...) :
  cannot remove file 'C:\Users\SBASHE~1\AppData\Local/LTMRdata-test/LTMRdata-test/Cache/fish/Taxa=Acipenser/part-0.parquet', reason 'Permission denied'
4: In FUN(X[[i]], ...) :
  cannot remove file 'C:\Users\SBASHE~1\AppData\Local/LTMRdata-test/LTMRdata-test/Cache/fish/Taxa=Actinopteri/part-0.parquet', reason 'Permission denied'
5: In FUN(X[[i]], ...) :
  cannot remove file 'C:\Users\SBASHE~1\AppData\Local/LTMRdata-test/LTMRdata-test/Cache/fish/Taxa=Aequorea/part-0.parquet', reason 'Permission denied'
6: In FUN(X[[i]], ...) :
  cannot remove file 'C:\Users\SBASHE~1\AppData\Local/LTMRdata-test/LTMRdata-test/Cache/fish/Taxa=Allosmerus elongatus/part-0.parquet', reason 'Permission denied'
7: In FUN(X[[i]], ...) :
  cannot remove file 'C:\Users\SBASHE~1\AppData\Local/LTMRdata-test/LTMRdata-test/Cache/fish/Taxa=Alopias vulpinus/part-0.parquet', reason 'Permission denied'
8: In FUN(X[[i]], ...) :
  cannot remove file 'C:\Users\SBASHE~1\AppData\Local/LTMRdata-test/LTMRdata-test/Cache/fish/Taxa=Alosa sapidissima/part-0.parquet', reason 'Permission denied'```

**Comment by jeanetteclark** [comment]
this is really a bug in `arrow`, ticket filed here: https://issues.apache.org/jira/browse/ARROW-16421

I'll work on creating a more informative error message

**Comment by sbashevkin** [comment]
I think the best option would be to use `getOption("arrow_duck_con")` to know whether a table has been opened, and then have `clear_cache` error and tell users to restart R if so (and only for Windows users)

**Comment by sbashevkin** [comment]
This has now been fixed by `arrow` v8.0.0, which we now depend on since https://github.com/Delta-Stewardship-Council/deltafish/commit/47e98ee911951d700cc6b356cc718270b6c45b38. It probably can't hurt to retain the trycatch and warning messages just in case it crops up again.
